features/exploredash/src/main/java/org/dash/wallet/features/exploredash/ui/ctxspend/dialogs/GiftCardDetailsViewModel.kt

```diff
@@ -233,25 +234,16 @@ class GiftCardDetailsViewModel @Inject constructor(
         logOnPurchaseEvents(giftCard)
     }
 
-    private fun saveBarcode(barcodeUrl: String) {
+    private fun saveBarcode(giftCardNumber: String) {
         applicationScope.launch {
             try {
-                val result = Constants.HTTP_CLIENT.get(barcodeUrl)
-                require(result.isSuccessful && result.body != null) { "call is not successful" }
-                val bitmap = result.body!!.decodeBitmap()
-                val decodeResult = Qr.scanBarcode(bitmap)
-
-                if (decodeResult != null) {
-                    metadataProvider.updateGiftCardBarcode(
-                        transactionId,
-                        decodeResult.first.replace(" ", ""),
-                        decodeResult.second
-                    )
-                } else {
-                    log.error("ScanBarcode returned null: $barcodeUrl")
-                }
+                metadataProvider.updateGiftCardBarcode(
+                    transactionId,
+                    giftCardNumber.replace(" ", ""),
+                    BarcodeFormat.CODE_128 // Assuming CTX barcodes are all CODE_128
+                )
```

Comment:
I noticed under the current system (reading the barcode provided by CTX) that Amazon Barcodes didn't include the `-` (dashes).  `ABCD-EFGHJ-XYZ` was `ABCDEFGHJXYZ`

For GameStop, i noticed that the barcode also included the pin code.

Reply:
Removed dashes. No changes related to PIN per our discussion.